File: algorithms/c/27.c
#include <stdio.h>

int removeElement(int* nums, int numsSize, int val)
{
    int read, write;
    write = 0;
    for ( read = 0; read < numsSize; read++ )
        if ( nums[read] != val )
            nums[write++] = nums[read];
    return write;
}

int main()
{
    return 0;
}

File: algorithms/rust/27.rs
impl Solution {
    #[allow(unused)]
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut idx = 0;
        for i in 0..nums.len() {
            if nums[i] != val {
                nums[idx] = nums[i];
                idx += 1;
            }
        }
        nums.resize(idx, 0);
        idx as i32
    }
}

#[allow(dead_code)]
struct Solution;

fn main() {
    println!("Hello, world!");
}
